### PR TITLE
Maintain strapi-provider-email-nodemailer

### LINKED ISSUE #6998: Please maintain a SMTP email provider for better developer experience
- [x] **I have created my request on the Product Board before I submitted this issue**
- [ ] **I have looked at all the other requests on the Product Board before I submitted this issue**

**Please describe your feature request:**

I can see that you maintain several proprietary email providers including
Mailgun, Sendgrid and Amazon:
https://github.com/strapi/strapi/tree/master/packages
 As a default, you use sendmail. I plead in favour of maintaining a SMTP capable email provider.

There are two main advantages:
* SMTP is better for data privacy and GDPR-compliance
* There is now a ton of unmaintained or outdated email providers for SMTP:
  https://www.npmjs.com/search?q=strapi-provider-email-
  Therefore it would improve the developer experience a lot to avoid this
  confusion.

It took me a whole day to get my head through all the version conflicts and I
have now sent a PR to a V3-compatible fork of `strapi-provider-email-nodemailer`:
https://github.com/Saunved/strapi-provider-email-nodemailer/pull/1

I have requested that the npm package with the most obvious name should merge
this v3-compatible fork and release a new major version:
https://github.com/yutikom/strapi-provider-email-nodemailer/issues/9

The ideal solution in my opinion is that the Strapi community maintains a
nodemailer provider out-of-the-box just as it does for the other proprietary
email-providers, already. You can see here, it's not too much code to maintain:
https://github.com/roschaefer/strapi-provider-email-nodemailer/blob/master/lib/index.js

The effort is small and the benefit would be to save a lot of people a lot of
headache.

Best,
Robert

### PR DESCRIPTION
**Pull request by roschaefer**
### What does it do?

This will merge repository https://github.com/yutikom/strapi-provider-email-nodemailer/ into this mono repo under
```
packages/strapi-provider-email-nodemailer
```

~I followed this [suggestion](https://stackoverflow.com/a/61450995) and ran the following commands:~

### Why is it needed?

In summer I opened [issue 6998](https://github.com/strapi/strapi/issues/6998) because I had a terrible developer experience from a multitude of unmaintained or outdated 3rd party libraries for email delivery through SMTP. As explained in the issue, we made `strapi-provider-email-nodemailer` compatible with version 3.

I think the best idea would be to incorporate the library into the main strapi mono repository, so it does not get out of sync and stays maintained. Strapi is already maintaining several proprietary email providers including Mailgun, Sendgrid and Amazon: https://github.com/strapi/strapi/tree/master/packages

So why not SMTP? SMTP is not proprietary and it's better for data privacy and GDPR. Developer experience gets improved because less headache but also because most people have an email account already.

See the isssue for more details.

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/6998

### TODOs
- [ ] @yutikom needs to make @strapi a package owner of https://www.npmjs.com/package/strapi-provider-email-nodemailer
- [ ] update CI/CD pipeline of strapi to push the package to npm

close #6998

### CODE REVIEW

**Comment by roschaefer** [comment]
I'm running into merge conflicts when trying to sign-off the commits as described here:
https://github.com/strapi/strapi/runs/1299535416

```
git rebase HEAD~47 --signoff
Auto-merging yarn.lock
Auto-merging packages/strapi-utils/lib/sanitize-entity.js
CONFLICT (content): Merge conflict in packages/strapi-utils/lib/sanitize-entity.js
CONFLICT (add/add): Merge conflict in packages/strapi-utils/lib/content-types.js
Auto-merging packages/strapi-utils/lib/content-types.js
CONFLICT (add/add): Merge conflict in packages/strapi-utils/lib/__tests__/content-types.test.js
Auto-merging packages/strapi-utils/lib/__tests__/content-types.test.js
Auto-merging packages/strapi-plugin-upload/services/Upload.js
Auto-merging packages/strapi-database/package.json
Auto-merging packages/strapi-connector-mongoose/lib/mount-models.js
CONFLICT (content): Merge conflict in packages/strapi-connector-mongoose/lib/mount-models.js
Auto-merging packages/strapi-connector-bookshelf/lib/mount-models.js
CONFLICT (content): Merge conflict in packages/strapi-connector-bookshelf/lib/mount-models.js
error: could not apply e29c1b7de... add published_at (#7374)
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply e29c1b7de... add published_at (#7374)
```

**Comment by derrickmehaffy** [comment]
There are going to need to be a few changes required before we can merge this, I'll do a review sometime today to point them out.

File: packages/strapi-provider-email-nodemailer/.release-it.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/.release-it.json:1]: This file needs to be removed, we don't use `release-it`

File: packages/strapi-provider-email-nodemailer/LICENSE.md
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/LICENSE.md:1]: This needs to be changed over to our license, see: https://github.com/strapi/strapi/blob/master/packages/strapi-provider-email-amazon-ses/LICENSE as an example

File: packages/strapi-provider-email-nodemailer/README.md
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/README.md:1]: README needs to be reconstructed over to the similar providers:
https://github.com/strapi/strapi/blob/master/packages/strapi-provider-email-amazon-ses/README.md
https://github.com/strapi/strapi/blob/master/packages/strapi-provider-email-mailgun/README.md
https://github.com/strapi/strapi/blob/master/packages/strapi-provider-email-sendmail/README.md
as examples

File: packages/strapi-provider-email-nodemailer/package-lock.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package-lock.json:1]: Delete this file please, we use yarn for the mono-repo

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package.json:31]: Need to add maintainers block: https://github.com/strapi/strapi/blob/dc0a91c7be8f7e515617459875fcb865c0f68300/packages/strapi-provider-email-amazon-ses/package.json#L20-L30 for an example

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package.json:46]: Change this to the Strapi link

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package.json:48]: No idea where this came from but it can be removed

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package.json:20]: change to strapi one

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package.json:14]: remove scripts

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package.json:10]: @alexandrebodin I don't think we need dotenv here anymore right?

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package.json:51]: change to strapi one

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [inline packages/strapi-provider-email-nodemailer/package.json:52]: remove this, as we don't need release-it

**Review by derrickmehaffy — CHANGES_REQUESTED** [review]
We will need the npm page transferred to us before merging

**Comment by derrickmehaffy** [comment]
Also as a side note, there is a ton of messy commits in the history. If possible, it would be best to rebase this and get rid of all the extra commits.

**Comment by roschaefer** [comment]
@Saunved @yutikom @veitbjarsch @tjakubowski I'm OK to do all the changes requested by @derrickmehaffy  :point_up: but if I squash the commit history, then I will "own" the changes and you're not going to end up as [contributors](https://github.com/strapi/strapi/graphs/contributors) :crying_cat_face: 

Opinions?

**Comment by derrickmehaffy** [comment]
> @Saunved @yutikom @veitbjarsch @tjakubowski I'm OK to do all the changes requested by @derrickmehaffy  :point_up: but if I squash the commit history, then I will "own" the changes and you're not going to end up as [contributors](https://github.com/strapi/strapi/graphs/contributors) :crying_cat_face: 
> 
> Opinions?

I'll let @alexandrebodin confirm, but I think even when we merge we also squash.

**Comment by Saunved** [comment]
> @Saunved @yutikom @veitbjarsch @tjakubowski I'm OK to do all the changes requested by @derrickmehaffy  but if I squash the commit history, then I will "own" the changes and you're not going to end up as [contributors](https://github.com/strapi/strapi/graphs/contributors) 
> 
> Opinions?

Hi Robert. Thanks for undertaking this task and keeping us informed about it from time to time. I got an email because of your "@" mention, so thanks a lot for that.

As per [this Github blog article](https://github.blog/2018-01-29-commit-together-with-co-authors/) link, it is possible to attribute co-authors to a commit when you're squashing it, so all of us can be contributors to the repo.

Also, @derrickmehaffy mentioned that the original npm package needs to be transferred over before it is published. I believe @yutikom will have to be informed about this.

**Comment by veitbjarsch** [comment]
If @yutikom is okay with it [and it looks like he is](https://github.com/strapi/strapi/issues/6998#issuecomment-714803495) and somebody tells me how to transfer the npm package I can do this too.

**Comment by derrickmehaffy** [comment]
> As per this Github article, it is possible to attribute co-authors to a commit when you're squashing it, so all of us can be contributors to the repo.

Yup exactly (we don't want to cut any of you out as being listed, and want to intentionally leave your names in the package.json as the original author and contributors there also)

File: packages/strapi-provider-email-nodemailer/LICENSE
Comment (roschaefer) [inline packages/strapi-provider-email-nodemailer/LICENSE:1]: I replaced the previous [License](https://github.com/yutikom/strapi-provider-email-nodemailer/blob/master/LICENSE.md) with the one I found [here](https://github.com/strapi/strapi/blob/master/packages/strapi-provider-email-mailgun/LICENSE).

File: packages/strapi-provider-email-nodemailer/README.md
Comment (roschaefer) [inline packages/strapi-provider-email-nodemailer/README.md:133]: Something magical happened here :star2:  it seems that the linter of strapi has auto-indented this table and even javascript-linted the inline code above :point_up:

**Review by roschaefer — COMMENTED** [review]
@derrickmehaffy I believe I did all the requested changes, can you confirm?

Everyone except @yutikom is indeed listed as co-author of the commit, that's amazing and something new I learned, thank you @Saunved!

@yutikom I used `git shortlog -sne` I guess Github cannot assign the email address listed there.

File: packages/strapi-provider-email-nodemailer/README.md
Comment (veitbjarsch) [reply packages/strapi-provider-email-nodemailer/README.md:133]: awesome 👍

**Review by derrickmehaffy — APPROVED** [review]
LGTM

**Comment by derrickmehaffy** [comment]
I think all we need is to move the https://www.npmjs.com/package/strapi-provider-email-nodemailer over to the `strapi` org then we should be good to merge and include it in the next release.

@strapi/maintainers is there anything we are missing?

File: packages/strapi-provider-email-nodemailer/package.json
Comment (MattieBelt) [inline packages/strapi-provider-email-nodemailer/package.json:15]: Shouldn't it be either `git://` or `https://`.

File: packages/strapi-provider-email-nodemailer/package.json
Comment (MattieBelt) [inline packages/strapi-provider-email-nodemailer/package.json:3]: `v3.2.5`

**Review by MattieBelt — COMMENTED** [review]
Missing some couple of other keys in the `package.json` (e.g. `strapi`, `engines` etc.) this could be an example: https://github.com/strapi/strapi/blob/master/packages/strapi-provider-email-sendmail/package.json. Not sure what must be included (or in any specific order) by the teams opinion.

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [reply packages/strapi-provider-email-nodemailer/package.json:3]: We have a script to handle the versioning, but yeah probably not a bad idea to set it to the current version so the script picks it up

**Comment by veitbjarsch** [comment]
It wasn't possible for me to transfer the package to an organisation. It seems to be more difficult to change from a user to an organisation. I've wrote to the npm support. May they can help us.

**Comment by derrickmehaffy** [comment]
Spoke with @alexandrebodin for the package transfer it seems we need to send it to a person who can add it to our org. If you could transfer it to https://www.npmjs.com/~alexandrebodin we can get it moved :)

**Comment by veitbjarsch** [comment]
@derrickmehaffy @alexandrebodin I can add you to the npm package, but it may still not work. Here is the response of the npm support:

> Unfortunately, we are unable to transfer unscoped packages to an org. In the npm registry, unscoped packages are managed differently than scoped ones, and currently our "Orgs" (Teams) plans are meant to only provide scoped package management.

**Comment by alexandrebodin** [comment]
@roschaefer Thank you I'll be checking that soon

File: packages/strapi-provider-email-nodemailer/package.json
Comment (roschaefer) [reply packages/strapi-provider-email-nodemailer/package.json]: Done. Is it common to use the same version for all packages in a mono-repo? What would have happened if the library had a version `> 3.2.5` already?

File: packages/strapi-provider-email-nodemailer/package.json
Comment (roschaefer) [reply packages/strapi-provider-email-nodemailer/package.json]: Yes, thanks, according to https://docs.npmjs.com/cli/v6/configuring-npm/package-json#repository this is just an ordinary URL for humans.

Comment (roschaefer) [reply packages/strapi-provider-email-nodemailer/package.json]: What about the missing keys? I removed key `strapi` because of this comment: https://github.com/strapi/strapi/pull/8459#discussion_r512280074

Comment (MattieBelt) [reply packages/strapi-provider-email-nodemailer/package.json]: All the provider do have the `isProvider` except for `strapi-provider-upload-rackspace` & `strapi-provider-email-amazon-ses`, but I guess that this is something old and not used anymore. @derrickmehaffy?

**Comment by alexandrebodin** [comment]
@veitbjarsch I'll need you to add me as a maintainer on the npm pacakge so I can move it to the right organisation for future releases too :)

**Comment by veitbjarsch** [comment]
@alexandrebodin I've already added you as a maintainer to the package (as I wrote 4 days ago).
Screenshot 2020-11-02 at 19 00 55

**Comment by alexandrebodin** [comment]
Thanks so strange I could see it. now it is alright

**Review by derrickmehaffy — DISMISSED** [review]
In terms of readme and git history, everything looks good to me. Pending a review and approval from eng team.

File: packages/strapi-provider-email-nodemailer/package.json
Comment (derrickmehaffy) [reply packages/strapi-provider-email-nodemailer/package.json]: I've never noticed it before tbh @MattieBelt but worst case we can make our own PRs to fix any issues before release. Don't want to stall this PR just for minor stuff ;)

**Comment by alexandrebodin** [comment]
Will merge it when ready to republish a new version

**Review by petersg83 — APPROVED** [review]
LGTM

**Comment by alexandrebodin** [comment]
Hi @roschaefer All good with the new publish 👍  thank you for this great contribution !

**Pull request merged**
